### PR TITLE
Fix currency filter and country filter

### LINKED ISSUE #187: Contract currency filter not working
### Environment
* Python version:  3.7.16
* Nautobot version:  1.5.7
* nautobot-plugin-device-lifecycle-mgmt version:  1.1.2

### Expected Behavior
To be able to filter Contract by currency

   &nbsp;- currency
   &nbsp;&nbsp;- Unknown filter field

### Steps to Reproduce
1. Go to Contract List View (i.e. on [demo site](https://demo.nautobot.com/plugins/nautobot-device-lifecycle-mgmt/contract/))
2. Click Filter button
3. Choose any currency (or just let it default to the first one)
4. Click Apply

### PR DESCRIPTION
**Pull request by zackt25**
This will enable the multi selection for filtering in:

`currency` in the Contracts view
`country` in the Vendors view

Before you were only able to filter on 1 selection in regards to currency and country. Now you can select multiple and it will filter out.

This will close #187

### CODE REVIEW

File: nautobot_device_lifecycle_mgmt/forms.py [hunk 1]
```diff
@@ -913,11 +919,7 @@ class ProviderLCMFilterForm(BootstrapMixin, forms.ModelForm):
 
     q = forms.CharField(required=False, label="Search")
     name = forms.CharField(required=False)
-    country = forms.ChoiceField(
-        widget=StaticSelect2,
-        required=False,
-        choices=add_blank_choice(CountryCodes.CHOICES),
-    )
+    country = forms.MultipleChoiceField(required=False, choices=CountryCodes.CHOICES, widget=StaticSelect2Multiple())
```
Comment (itdependsnetworks) [inline nautobot_device_lifecycle_mgmt/forms.py:922]: why remove `add_blank_choice`?

Comment (zackt25) [reply nautobot_device_lifecycle_mgmt/forms.py]: I copied and pasted something with fixed and that wasn't in there. I added it back.

**Comment by zackt25** [comment]
It was agreed from now on for bug fixes we will support python 3.8 and 3.9. I've removed 3.7 from the CI file and changed the toml file to be higher than 3.8.

File: nautobot_device_lifecycle_mgmt/forms.py (hunk 1 above)
Comment (progala) [reply nautobot_device_lifecycle_mgmt/forms.py]: @itdependsnetworks I don't think `add_blank_choice` does anything when used with `MultipleChoiceField`? We previously had `ChoiceField` here hence why `add_blank_choice` was used.

Comment (zackt25) [reply nautobot_device_lifecycle_mgmt/forms.py]: I tested both ways and there isn't a noticeable difference when selecting the filter.

Comment (itdependsnetworks) [reply nautobot_device_lifecycle_mgmt/forms.py]: ok, makes sense, yet another pylint nautobot use case to consider.

**Pull request merged**